### ISSUE TITLE
`age` field in /messages duplicated as top-level field

### ISSUE THREAD

**Issue by joepie91** [matrix-org/synapse#7925]
When calling the `/messages` endpoint, each event in the `chunk` will have a top-level `age` property that duplicates the one within `unsigned`. The top-level property is not a part of the spec.

**Comment by joepie91** [comment]
Same happens for `replaces_state` (related: matrix-org/matrix-doc#1167). Does the Synapse team prefer separate issues for every incorrect field, or is it okay to group them together by "type of error" like this?

**Comment by clokep** [comment]
@joepie91 I think grouping them together is OK, as long as it is clear what properties are incorrect on what endpoints.

Hopefully most of these are easy to fix, as long as clients aren't depending on them.

**Comment by richvdh** [comment]
>  as long as clients aren't depending on them.

well, therein lies the rub.

**Comment by joepie91** [comment]
Another case: `redacted_because`.

**Comment by joepie91** [comment]
> I don't think the unsigned.age property is part of the spec either, is it?

Yeah, I think you're correct. I'm not sure why I phrased it like that.

**Comment by richvdh** [comment]
> > as long as clients aren't depending on them.
> 
> well, therein lies the rub.

Thinking about this a bit harder: any client that wants to use these properties, and be compatible with `/sync`, must be using the values from `unsigned` rather than the top-level properties... so I'd be *reasonably* confident removing them from the top level.

**Comment by richvdh** [comment]
For anyone thinking of looking at this: the code that does the copying is at https://github.com/matrix-org/synapse/blob/v1.43.0/synapse/events/utils.py#L255-L265. Note that that code also includes `prev_content` which is a slightly more complex case (see https://github.com/matrix-org/matrix-doc/issues/877). I'd also be inclined to get rid of the unspecced `user_id` property.

**Comment by richvdh** [comment]
> any client that wants to use these properties, and be compatible with /sync, must be using the values from unsigned rather than the top-level properties... so I'd be reasonably confident removing them from the top level.

ARGH. Of course that neglects application services, which don't need to `/sync` (and the AS API copies the keys to the top level)

**Comment by richvdh** [comment]
I still think we should work to get rid of these spurious top-level keys, but it seems like there is potential for breakage. Accordingly it might be best for us to clarify the situation for `prev_content` first (https://github.com/matrix-org/matrix-doc/issues/877) so that then we can break everything at once.

**Comment by richvdh** [comment]
[MSC3442](https://github.com/matrix-org/matrix-doc/blob/main/proposals/3442-move-prev-content.md) clarified that `prev_content` should be returned under `unsigned`. 

It's still possible that some things - most likely ASes which don't need to worry about `/sync` - will be broken by removal of these properties, but we really should try and clean them up anyway.
